diff -ru apache-couchdb-0.10.1/src/couchdb/couch_rep.erl apache-couchdb-0.10.1/src/couchdb/couch_rep.erl
--- apache-couchdb-0.10.1/src/couchdb/couch_rep.erl	2009-09-25 17:51:12.000000000 -0700
+++ apache-couchdb-0.10.1/src/couchdb/couch_rep.erl	2009-10-17 13:09:53.000000000 -0700
@@ -28,6 +28,7 @@
     source,
     target,
     continuous,
+    designonly,
     init_args,
     checkpoint_scheduled = nil,
 
@@ -102,6 +103,7 @@
     TargetProps = proplists:get_value(<<"target">>, PostProps),
 
     Continuous = proplists:get_value(<<"continuous">>, PostProps, false),
+    DesignOnly = proplists:get_value(<<"designonly">>, PostProps, false),
 
     Source = open_db(SourceProps, UserCtx),
     Target = open_db(TargetProps, UserCtx),
@@ -143,6 +145,7 @@
         source = Source,
         target = Target,
         continuous = Continuous,
+        designonly = DesignOnly,
         init_args = InitArgs,
         stats = Stats,
         checkpoint_scheduled = nil,
@@ -388,7 +391,13 @@
     false ->
         ""
     end,
-    {Base, Extension}.
+    Extension2 = case proplists:get_value(<<"designonly">>, Props, false) of
+    true ->
+        "+designonly";
+    false ->
+        ""
+    end,
+    {Base, Extension ++ Extension2}.
 
 maybe_add_trailing_slash(Url) ->
     re:replace(Url, "[^/]$", "&/", [{return, list}]).
--- apache-couchdb-0.10.1/src/couchdb/couch_rep_reader.erl	2009-10-05 22:47:27.000000000 -0700
+++ apache-couchdb-0.10.1/src/couchdb/couch_rep_reader.erl	2009-10-17 13:37:00.000000000 -0700
@@ -49,7 +49,7 @@
 next(Pid) ->
     gen_server:call(Pid, next_docs, infinity).
 
-init([Parent, Source, MissingRevs, _PostProps]) ->
+init([Parent, Source, MissingRevs, PostProps]) ->
     process_flag(trap_exit, true),
     if is_record(Source, http_db) ->
         #url{host=Host, port=Port} = ibrowse_lib:parse_url(Source#http_db.url),
@@ -57,7 +57,7 @@
         ibrowse:set_max_pipeline_size(Host, Port, ?MAX_PIPELINE_SIZE);
     true -> ok end,
     Self = self(),
-    ReaderLoop = spawn_link(fun() -> reader_loop(Self, Source, MissingRevs) end),
+    ReaderLoop = spawn_link(fun() -> reader_loop(Self, Source, MissingRevs, proplists:get_value(<<"designonly">>, PostProps, false)) end),
     State = #state{
         parent = Parent,
         source = Source,
@@ -234,20 +234,27 @@
     end,
     [Transform(Result) || Result <- JsonResults].
 
-reader_loop(ReaderServer, Source, MissingRevsServer) ->
+reader_loop(ReaderServer, Source, MissingRevsServer, DesignOnly) ->
     case couch_rep_missing_revs:next(MissingRevsServer) of
     complete ->
         exit(complete);
     {HighSeq, IdsRevs} ->
         % to be safe, make sure Results are sorted by source_seq
-        SortedIdsRevs = lists:keysort(2, IdsRevs),
+        UnfilteredSortedIdsRevs = lists:keysort(2, IdsRevs),
+        SortedIdsRevs = case DesignOnly of
+            true ->
+                lists:filter(fun({Id,_,_}) -> (Id >= <<"_design/">>) and (Id < <<"_design/",255>>) end, UnfilteredSortedIdsRevs);
+            false ->
+                UnfilteredSortedIdsRevs
+        end,
+
         RequestSeqs = [S || {_,S,_} <- SortedIdsRevs],
         gen_server:call(ReaderServer, {add_request_seqs, RequestSeqs}, infinity),
         case Source of
         #http_db{} ->
             [gen_server:call(ReaderServer, {open_remote_doc, Id, Seq, Revs},
                 infinity) || {Id,Seq,Revs} <- SortedIdsRevs],
-            reader_loop(ReaderServer, Source, MissingRevsServer);
+            reader_loop(ReaderServer, Source, MissingRevsServer, DesignOnly);
         _Local ->
             Source2 = maybe_reopen_db(Source, HighSeq),
             lists:foreach(fun({Id,Seq,Revs}) ->
@@ -256,7 +263,7 @@
                 gen_server:call(ReaderServer, {add_docs, Seq, JustTheDocs},
                     infinity)
             end, SortedIdsRevs),
-            reader_loop(ReaderServer, Source2, MissingRevsServer)
+            reader_loop(ReaderServer, Source2, MissingRevsServer, DesignOnly)
         end
     end.
 
